src/main/java/com/hrznstudio/sandbox/network/Packet.java

```diff
@@ -0,0 +1,13 @@
+package com.hrznstudio.sandbox.network;
+
+import net.minecraft.util.PacketByteBuf;
+
+import java.io.IOException;
+
+public interface Packet {
+    void read(PacketByteBuf var1) throws IOException;
+
+    void write(PacketByteBuf var1) throws IOException;
+
+    void apply();
```

Comment:
question: why does apply() method not also provide the context for scheduling packets on the receiving side?

Reply:
Because they're always executed on the main thread for the receiving side anyway